autogpt/commands/file_operations.py

```diff
@@ -179,6 +194,59 @@ def append_to_file(filename: str, text: str, should_log: bool = True) -> str:
         return f"Error: {str(e)}"
 
 
+@command(
+    "update_file",
+    "Update file",
+    '"filename": "<filename>", '
+    '"old_text": "<old_text>", "new_text": "<new_text>", '
+    '"match_index": "<match_index>"',
+)
+def update_file(filename: str, old_text: str, new_text: str, match_index=None):
+    """Update a file by replacing one specific occurrence of old_text with new_text or replacing all occurrences
+    this method excludes the end-of-line character at the end of old_text and new_text and uses a directory variable to
+    prevent auto-gpt from using the wrong directory
+
+    Args:
+        directory (str): directory where the file is located
+        filename (str): The name of the file
+        old_text (str): String to be replaced   \n will be stripped from string
+        new_text (str): New string   \n will be stripped from string
+        match_index (str): indexed match where the update should take place, takes (none, all and numbers)
+
+    """
+    try:
+        with open(filename, "r", encoding="utf-8") as f:
+            content = f.read()
+
+        old_text = old_text.rstrip("\n")
+        new_text = new_text.rstrip("\n")
```

Comment:
this breaks multi-line replacements

Reply:
> Changes to `check_duplicate_operation` and `log_operation` are out of scope, please revert as there are other PRs targeting issues with that feature

Not possible as update_file needs the extra vriable to be able to avoid "file already written" errors from duplicate_operations which in turn requires the log_operation to be refitted as well